### ISSUE TITLE
New deferred intent UPE compatibility with WooPay

### ISSUE THREAD

**Issue by FangedParakeet** [Automattic/woocommerce-payments#6291]
### Description

Currently, the split-UPE achieves compatibility with WooPay by falling back to the legacy card element for the CC gateway when WooPay is enabled. With the deferred intent UPE, this is no longer a necessity, since we will create the payment method prior to confirming the intent, allowing us to perform the regular WooPay cloning procedures in time.

This issue is to add WooPay compatibility to the deferred intent UPE card payment method. At this point, when the deferred intent UPE is enabled, we will no longer need to include any legacy Stripe scripts and we can present a UPE payment method in all available payment gateways. Consequently, we will need to amend the code where gateways and scripts are registered to reflect this.

As discussed in our roundtable, we would also like this issue to cover all available WooPay flows, so this should support an implementation in both the shortcode and blocks checkouts. 

### Acceptance criteria

* When WooPay is enabled, UPE card payment method remains.
* WooPay registration and checkout are both functional.
* WooPay is implemented on both blocks and shortcode checkouts.

### Dev notes

We should to reuse as much as possible from the existing legacy card gateway implementation, but from this vantage point I cannot yet see how much will be reusable. 

### Additional context

Blocked by #5020

**Comment by FangedParakeet** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Heisenberg-63586f2ec60359001310fa6a/issues/Automattic/woocommerce-payments/6291?planning-poker) with Zenhub @mdmoore @timur27

**Comment by FangedParakeet** [comment]
Blocked by #6393.
